prover/core/prover.go

```diff
@@ -42,14 +42,14 @@ func NewProverCore(cfg *config.ProverCoreConfig) (*ProverCore, error) {
 
 	var vk string
 	var rawVK *C.char
-	defer C.free(unsafe.Pointer(rawVK))
 	if cfg.ProofType == message.ProofTypeBatch {
 		C.init_batch_prover(paramsPathStr, assetsPathStr)
 		rawVK = C.get_batch_vk()
 	} else if cfg.ProofType == message.ProofTypeChunk {
 		C.init_chunk_prover(paramsPathStr, assetsPathStr)
 		rawVK = C.get_chunk_vk()
 	}
+	defer C.free_c_chars(rawVK)
```

Comment:
the rawVK is newed in go code, so it should be released in `defer C.free(unsafe.Pointer(rawVK))`?